sonar-db/src/main/resources/org/sonar/db/user/RoleMapper.xml

```diff
@@ -20,6 +20,24 @@
     </where>
   </select>
 
+  <select id="selectUserPermissionsByPermissionAndUserId" parameterType="map" resultType="UserRole">
```

Comment:
this method should only return component id, as permission and user and given in parameter.
So it could be called selectComponentsByPermissionAndUserId